client/components/ResponsiveFlex/ResponsiveFlex.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { PropsWithChildren, useEffect } from 'react';
+
+interface Props {
+  breakpoint: number;
+}
+
+function ResponsiveFlex({ breakpoint, children }: PropsWithChildren<Props>) {
+  const containerRef = React.createRef<HTMLDivElement>();
+
+  useEffect(() => {
+    if (breakpoint) {
+      if (document.head.querySelector(`[data-flex-breakpoint="${breakpoint}"]`)) {
+        containerRef.current.className = `responsive-flex--${breakpoint}`;
+      } else {
+        const styleEl = document.createElement('style');
+        styleEl.setAttribute('data-flex-breakpoint', `${breakpoint}`);
+        styleEl.innerHTML = `
+        .responsive-flex--${breakpoint} {
+          display: flex;
+          flex-direction: row;
+        }
+        @media screen and (max-width: ${breakpoint}px) {
+          .responsive-flex--${breakpoint} {
+            flex-direction: column;
+          }
+          .responsive-flex--${breakpoint} div {
+            width: 100%;
+            margin-left: 0;
+            margin-right: 0;
+          }
+        }`;
+        document.head.appendChild(styleEl);
+        containerRef.current.className = `responsive-flex--${breakpoint}`;
+      }
+    }
+  }, [containerRef]);
```

Comment:
Why do we have the `containerRef` as a dependency for `useEffect`?

I think we need to have the `breakpoint`, no?